packages/accounts/src/decode-account.ts

```diff
@@ -28,3 +28,41 @@ export function decodeAccount<TData extends object, TAddress extends string = st
         throw newError;
     }
 }
+
+function accountExists<TData extends object>(account: Account<TData> | MaybeAccount<TData>): account is Account<TData> {
+    return !('exists' in account) || ('exists' in account && account.exists);
+}
+
+/** Asserts that an account has been decoded. */
+export function assertAccountDecoded<TData extends object, TAddress extends string = string>(
+    account: Account<TData | Uint8Array, TAddress>,
+): asserts account is Account<TData, TAddress>;
+export function assertAccountDecoded<TData extends object, TAddress extends string = string>(
+    account: MaybeAccount<TData | Uint8Array, TAddress>,
+): asserts account is MaybeAccount<TData, TAddress>;
+export function assertAccountDecoded<TData extends object, TAddress extends string = string>(
+    account: Account<TData | Uint8Array, TAddress> | MaybeAccount<TData | Uint8Array, TAddress>,
+): asserts account is Account<TData, TAddress> | MaybeAccount<TData, TAddress> {
+    if (accountExists(account) && account.data instanceof Uint8Array) {
+        // TODO: coded error.
+        throw new Error(`Expected account [${account.address}] to be decoded.`);
+    }
+}
+
+/** Asserts that all accounts have been decoded. */
+export function assertAccountsDecoded<TData extends object, TAddress extends string = string>(
+    accounts: Account<TData | Uint8Array, TAddress>[],
+): asserts accounts is Account<TData, TAddress>[];
+export function assertAccountsDecoded<TData extends object, TAddress extends string = string>(
+    accounts: MaybeAccount<TData | Uint8Array, TAddress>[],
+): asserts accounts is MaybeAccount<TData, TAddress>[];
+export function assertAccountsDecoded<TData extends object, TAddress extends string = string>(
+    accounts: (Account<TData | Uint8Array, TAddress> | MaybeAccount<TData | Uint8Array, TAddress>)[],
+): asserts accounts is (Account<TData, TAddress> | MaybeAccount<TData, TAddress>)[] {
+    const encoded = accounts.filter(a => accountExists(a) && a.data instanceof Uint8Array);
+    if (encoded.length > 0) {
+        const encodedAddresses = encoded.map(a => a.address).join(', ');
+        // TODO: Coded error.
+        throw new Error(`Expected accounts [${encodedAddresses}] to be decoded.`);
+    }
+}
```

Comment:
I hate how TypeScript doesn't let you run an assertion function over all elements in an array to assert that the array contains * whatever it did before * but definitely not a `Uint8Array`.

If we could do that then you could pass _various_ accounts with _various_ data into this function and have them all come out clean (ie. without `Uint8Arrays` among them).

https://stackoverflow.com/a/77323905/802047